pkg/util/testing/wrappers.go

```diff
@@ -218,17 +218,15 @@ func (w *WorkloadWrapper) AdmissionChecks(checks ...kueue.AdmissionCheckState) *
 	return w
 }
 
-func (w *WorkloadWrapper) OwnerReference(gvk schema.GroupVersionKind, name, uid string, controller, blockDeletion bool) *WorkloadWrapper {
-	w.OwnerReferences = []metav1.OwnerReference{
-		{
-			APIVersion:         gvk.GroupVersion().String(),
-			Kind:               gvk.Kind,
-			Name:               name,
-			UID:                types.UID(uid),
-			Controller:         &controller,
-			BlockOwnerDeletion: &blockDeletion,
-		},
-	}
+func (w *WorkloadWrapper) OwnerReference(gvk schema.GroupVersionKind, name, uid string, controller, blockDeletion *bool) *WorkloadWrapper {
```

Comment:
Is this change necessary? This new approach leads to much more code.

Reply:
Not really, I reverted them by keeping the OwnerReference function.  For the need of my unit test I now introduce SetOwnerReference and SetControllerReference. These are analogs of the controllerutil functions, which either pass (nil, nil) or (true, true). So now, the changes are scoped to the TestReconcile unit test. If you like this approach I could have a follow up to eliminate the OwnerReference function and replace it with either of the two new functions.

Still, inside this test I add a bunch of "SetOwnerReference" and "SetControllerReference" lines. The alternatives I see:
1. have an extra param in the unit test like "verifyOwnerReferences" that would (1) skip the code of adding owner references, (2) use another set of cmpopts for the workload. 
2. have a dedicated unit test for this scenario, say TestReconcileWithOwnerReferences
3. do not add unit tests for owner references

IMO none of the alternatives is tempting. Indeed adding a bunch of lines seems like a lot of code, but easy to understand. WDYT?